src/pages/FeeRateTracker/FeeRateTrackerComp.tsx

```diff
@@ -6,47 +6,77 @@ import { useEffect, useMemo, useState } from 'react'
 import dayjs from 'dayjs'
 import styles from './styles.module.scss'
 import { isScreenSmallerThan1200 } from '../../utils/screen'
-import { useIsMobile } from '../../utils/hook'
+import i18n from '../../utils/i18n'
+import { useAppState } from '../../contexts/providers'
 
 export const FeeRateCards = ({ transactionFeeRates }: { transactionFeeRates: FeeRateTracker.TransactionFeeRate[] }) => {
+  const getWeightedMedian = (tfrs: FeeRateTracker.TransactionFeeRate[]): number => {
+    if (!tfrs || tfrs.length === 0) {
+      return 0
+    }
+    return tfrs.length % 2 === 0
+      ? (tfrs[tfrs.length / 2 - 1].confirmationTime + tfrs[tfrs.length / 2 - 1].confirmationTime) / 2
+      : tfrs[(tfrs.length - 1) / 2].confirmationTime
+  }
+
+  const allFrs = [...transactionFeeRates.filter(r => r.confirmationTime === 0 || r.confirmationTime)].sort(
+    (a, b) => a.confirmationTime - b.confirmationTime,
+  )
+  const avgConfirmationTime = getWeightedMedian(allFrs)
+  const lowFrs = allFrs.filter(r => r.confirmationTime >= avgConfirmationTime)
+  const lowConfirmationTime = getWeightedMedian(lowFrs)
+  const highFrs = allFrs.filter(r => r.confirmationTime <= avgConfirmationTime)
+  const highConfirmationTime = getWeightedMedian(highFrs)
+
   const feeRateCards = [
     {
-      priority: 'Low',
-      limitTimeDisplay: '3 mins',
-      limitTimeSeconds: 180,
+      priority: i18n.t('fee_rate_tracker.low'),
+      tfrs: lowFrs,
+      confirmationTime: lowConfirmationTime,
     },
     {
-      priority: 'Average',
-      limitTimeDisplay: '1 mins',
-      limitTimeSeconds: 60,
+      priority: i18n.t('fee_rate_tracker.average'),
+      tfrs: allFrs,
+      confirmationTime: avgConfirmationTime,
     },
     {
-      priority: 'High',
-      limitTimeDisplay: '30 secs',
-      limitTimeSeconds: 30,
+      priority: i18n.t('fee_rate_tracker.high'),
+      tfrs: highFrs,
+      confirmationTime: highConfirmationTime,
     },
   ].map(c => {
-    const frs = transactionFeeRates.filter(r => {
-      return r.confirmationTime === 0 || (r.confirmationTime && r.confirmationTime <= c.limitTimeSeconds)
-    })
     const feeRate =
-      frs.length === 0
+      c.tfrs.length === 0
         ? 0
-        : BigNumber.sum(...frs.map(r => r.feeRate))
-            .div(frs.length)
+        : BigNumber.sum(...c.tfrs.map(r => r.feeRate))
+            .div(c.tfrs.length)
             .dp(3, BigNumber.ROUND_HALF_EVEN)
             .toNumber()
-    return { ...c, feeRate } as FeeRateTracker.FeeRateCard
+    return {
+      priority: c.priority,
+      feeRate,
+      confirmationTime: c.confirmationTime,
+    } as FeeRateTracker.FeeRateCard
```

Comment:
Try to avoid using `as`, where you can put the type on top, something like this: `.map((c): FeeRateTracker.FeeRateCard => { ... }`